### PR TITLE
Don't propagate server.port

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes spring-cloud/spring-cloud-deployer-cloudfoundry#46

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 1]
```diff
@@ -15,10 +15,10 @@
  */
 package org.springframework.cloud.deployer.spi.cloudfoundry;
 
-import static java.lang.Integer.parseInt;
+import static java.lang.Integer.*;
```
Comment (markfisher) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: update formatting rules to avoid wildcard imports

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 2]
```diff
@@ -100,58 +100,67 @@ public String deploy(AppDeploymentRequest request) {
 		Map<String, String> envVariables = new HashMap<>();
 
 		try {
+			Map<String, String> properties = new HashMap<>(request.getDefinition().getProperties());
+			// Remove server.port as CF assigns a port for us, and we don't want to override that
+			String port = properties.remove("server.port");
+			if (port != null) {
+				logger.info(String.format("Ignoring server.port=%s property for app %s, as Cloud Foundry will assign a dynamic port", name, port));
```
Comment (markfisher) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: INFO might be fine, but should we consider WARN? cc @markpollack

**Comment by markfisher** [comment]
lots of formatting changes make this diff hard to follow... I assume the only actual changes are within lines 103-108?

**Comment by ericbottard** [comment]
> lots of formatting changes make this diff hard to follow... I assume the only actual changes are within lines 103-108?

yes.
Grrr... I just updated those (my formatting prefs) because Janne made me realize I lost them (on IDEA upgrade, I assume). I'm using the `spring-framework` scheme I always used. Maybe the code was formatted differently before.
Anyhow, I fixed the statics and force pushed

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java (hunk 2 above)
Comment (markpollack) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: I'd make it WARN since it is something one should ideally take action on to remove from the stream definition.

Comment (markpollack) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: The log message isn't quite right, the individual app will get a dynamic port, but the route to the app - which his what is exposed to the public - will be over port 80, we should indicate that instead.  Agreed?

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Addressed review comments

**Comment by ericbottard** [comment]
Closing in favor of https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/pull/78
